pkg/env/dev.go

```diff
@@ -256,6 +262,25 @@ func (d *dev) CreateARMResourceGroupRoleAssignment(ctx context.Context, fpAuthor
 		}
 	}
 
-	d.log.Print("development mode: refreshing authorizer")
-	return fpAuthorizer.(*refreshableAuthorizer).Refresh()
+	// Issue: https://github.com/Azure/ARO-RP/issues/31
+	// rbac client returns right permissions, but access is not yet propagated
+	// in the azure backends. We test by trying to call API directly and check if
+	// role was applied.
+	fpAuthorizer.(*refreshableAuthorizer).Refresh()
+	return wait.Poll(time.Second, time.Minute, func() (bool, error) {
+		// this should always error. Either 403 or 404
+		_, err := d.deployments.GetDeployment(ctx, oc.Properties.ResourceGroup, "dummy")
+		if detailedError, ok := err.(autorest.DetailedError); ok {
+			if requestError, ok := detailedError.Original.(azure.RequestError); ok &&
+				requestError.ServiceError != nil {
+				if requestError.ServiceError.Code == "AuthorizationFailed" {
+					return false, nil
+				}
+				if requestError.ServiceError.Code == "DeploymentNotFound" {
```

Comment:
not needed?

Reply:
either we say - all errors, expect 403 is Fine - yes. Or we want to be explicit...